src/test/java/common/bankarskiSistem/unit/service/ConversionServiceUnitTests.java

```diff
@@ -0,0 +1,284 @@
+        //When
+        Exception exception = assertThrows(NullPointerException.class,
+                () -> conversionService.updateConversion(conversion));
+
+        //Then
+        String expectedMessage = "Null conversion";
+        String actualMessage = exception.getMessage();
+        assertEquals(actualMessage, expectedMessage);
+    }
+
+    @ParameterizedTest
+    @MethodSource("common.bankarskiSistem.parametrized.ConversionServiceParameters#generateConversion")
+    public void whenUpdateNonExistingConversion_thenThrowNullPointerException(Conversion conversion) {
+        //Given
+        when(conversionRepository.findByIdConversion(conversion.getIdConversion()))
+                .thenReturn(Optional.empty());
+
+        //When
+        Exception exception = assertThrows(NullPointerException.class,
+                () -> conversionService.updateConversion(conversion));
+
+        //Then
+        String expectedMessage = "No such conversion exists!";
+        String actualMessage = exception.getMessage();
+        assertEquals(actualMessage, expectedMessage);
+    }
+
+    @ParameterizedTest
+    @MethodSource("common.bankarskiSistem.parametrized.ConversionServiceParameters#generateConversion")
+    public void whenUpdateExistingConversion_thenConversionIsReturned(Conversion conversion) {
+        //Given
+        when(conversionRepository.findByIdConversion(conversion.getIdConversion()))
+                .thenReturn(Optional.of(conversion));
+        when(conversionRepository.save(any(Conversion.class))).thenReturn(conversion);
+
+        //When
+        Conversion result = conversionService.updateConversion(conversion);
+
+        //Then
+        assertThat(result.getIdConversion()).isEqualTo(conversion.getIdConversion());
+    }
+
+    @ParameterizedTest
+    @MethodSource("common.bankarskiSistem.parametrized.ConversionServiceParameters#generateConversion")
+    public void whenUpdateConversionInExistingConversion_thenAllFieldsAreUpdated(Conversion conversionOld) {
+        //Given
+        int idConversion = conversionOld.getIdConversion();
+        Conversion conversionNew = new Conversion();
+        conversionNew.setIdConversion(idConversion);
+        double newValue = 200;
+        Currency newCurrencyFrom = Currency.EUR;
+        Currency newCurrencyTo = Currency.RSD;
+        conversionNew.setValue(newValue);
+        conversionNew.setCurrencyFrom(newCurrencyFrom);
+        conversionNew.setCurrencyTo(newCurrencyTo);
+        when(conversionRepository.findByIdConversion(idConversion)).thenReturn(Optional.of(conversionOld));
+        when(conversionRepository.save(any(Conversion.class))).thenReturn(conversionNew);
+
+        //When
+        Conversion resultConversion = conversionService.updateConversion(conversionOld);
+
+        //Then
+        assertThat(resultConversion.getIdConversion()).isEqualTo(idConversion);
```

Comment:
Mozes da napravis metodu (npr. updateConversion_params) u ConversionServiceParameters koja ce imati vise Arguments.of(). Tako ces moci da testiras vise slucajeva kada su u pitanju ove promjene.